x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -110,6 +110,16 @@ export function isSourceDataChartableForDetector(job: CombinedJob, detectorIndex
           }
         }
       }
+
+      // We also cannot plot the source data if they datafeed uses any field defined by runtime_mappings
+      const runtimeMappings =
```

Comment:
a lot of this code is duplicated, it might be worth creating a `hasRuntimeMappings` function which returns a boolean which is used in both of these functions?

Reply:
Updated here https://github.com/elastic/kibana/pull/86407/commits/0175531f097e142cc0c3c42a22e2d28ad25d68dc